tests/kernel/sched/schedule_api/src/test_sched_timeslice_reset.c

```diff
@@ -51,6 +51,7 @@ static void thread_time_slice(void *p1, void *p2, void *p3)
 {
 	uint32_t t = cycles_delta(&elapsed_slice);
 	uint32_t expected_slice_min, expected_slice_max;
+	uint32_t switch_tolerance = k_ms_to_ticks_ceil32(TASK_SWITCH_TOLERANCE);
```

Comment:
I think `TASK_SWITCH_TOLERANCE` should be renamed to add ` _TICKS` suffix, then

    #define TASK_SWITCH_TOLERANCE_MS k_ms_to_ticks_ceil32(TASK_SWITCH_TOLERANCE_TICKS);

rather than implying this needs to be done at runtime.

Reply:
> I'm a bit confused by this; I would expect a timeslice to be reset when the running thread becomes unrunnable; the next thread is entitled to a full timeslice. If the running thread is preempted the [documentation](https://docs.zephyrproject.org/latest/reference/kernel/scheduling/index.html?highlight=slice#preemptive-time-slicing) suggests the preempted thread should resume and only be given the remainder of its slice.
> 
> Maybe zephyr has a different concept of timeslice?

yep, that is confuse, I've confirmed with Andy that we don't do anything fancy and when a thread is preempted it gets the full slice. Will close this and fix the documentation.